### PR TITLE
Platform info

### PR DESCRIPTION
**Pull request by Cashmaney**
Small addition to display some extra info for group out of date and configuration needed statuses

### CODE REVIEW

File: cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs [hunk 1]
```diff
@@ -1,21 +1,30 @@
+#[cfg(feature = "SGX_MODE_HW")]
+use enclave_ffi_types::NodeAuthResult;
 ///
 /// These functions run off chain, and so are not limited by deterministic limitations. Feel free
 /// to go crazy with random generation entropy, time requirements, or whatever else
```
Comment (reuvenpo) [inline cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs:5]: please put this import below the docstring, with an empty line between them, and then replace the `///` with `//!` so it's correctly applied to the module

File: cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
+#[cfg(feature = "SGX_MODE_HW")]
+use crate::registration::report::AttestationReport;
 use crate::storage::write_to_untrusted;
 use crate::utils::{attest_from_key, validate_const_ptr, validate_mut_ptr, validate_mut_slice};
 
 use super::attestation::create_attestation_certificate;
+#[cfg(feature = "SGX_MODE_HW")]
+use super::cert::ocall_get_update_info;
+#[cfg(feature = "SGX_MODE_HW")]
+use super::cert::verify_quote_status;
```
Comment (reuvenpo) [inline cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs:27]: You can probably compress these imports that require `#[cfg(feature = "SGX_MODE_HW")]` under fewer clauses.
at least everything from this crate can be combined so you use fewer `cfg`s

File: cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs [hunk 3]
```diff
@@ -211,3 +222,84 @@ pub unsafe extern "C" fn ecall_key_gen(
...
+                    1,
+                    &mut update_info as *mut sgx_update_info_bit_t,
+                );
+
+                if res != sgx_status_t::SGX_SUCCESS {
+                    println!("res={:?}", res);
+                    return res;
+                }
+
+                if rt != sgx_status_t::SGX_SUCCESS {
```
Comment (reuvenpo) [inline cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs:276]: please write this as an unsafe function

File: cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs [hunk 4]
```diff
@@ -211,3 +222,84 @@ pub unsafe extern "C" fn ecall_key_gen(
...
+
+                    if update_info.ucodeUpdate != 0 {
+                        println!("Processor Firmware Update (ucodeUpdate). A security upgrade for your computing
+                            device is required for this application to continue to provide you with a high degree of
+                            security. Please contact your device manufacturer’s support website for a BIOS update
+                            for this system");
```
Comment (reuvenpo) [inline cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs:282]: Is the indentation omitted by the compiler?
I think you need to add `\` or something.

File: cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs [hunk 5]
```diff
@@ -211,3 +222,84 @@ pub unsafe extern "C" fn ecall_key_gen(
...
     sgx_status_t::SGX_SUCCESS
 }
+
+#[cfg(not(feature = "SGX_MODE_HW"))]
+fn verify_local_report(_cert: &[u8]) -> sgx_status_t {
+    sgx_status_t::SGX_SUCCESS
+}
+
+#[cfg(feature = "SGX_MODE_HW")]
+fn verify_local_report(cert: &[u8]) -> sgx_status_t {
```
Comment (reuvenpo) [inline cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs:232]: I prefer returning `SgxResult` if its not an extern function

File: cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs [hunk 6]
```diff
@@ -211,3 +222,84 @@ pub unsafe extern "C" fn ecall_key_gen(
...
+    match node_auth_result {
+        NodeAuthResult::GroupOutOfDate | NodeAuthResult::SwHardeningAndConfigurationNeeded => unsafe {
+            if let Some(platform_info) = report.platform_info_blob {
+                let mut update_info = sgx_update_info_bit_t::default();
+                let mut rt = sgx_status_t::default();
+                let res = ocall_get_update_info(
+                    &mut rt as *mut sgx_status_t,
+                    platform_info[4..].as_ptr() as *const sgx_platform_info_t,
+                    1,
+                    &mut update_info as *mut sgx_update_info_bit_t,
```
Comment (reuvenpo) [inline cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs:268]: I think you don't need the ` as *mut sgx_update_info_bit_t` bit

File: cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs (hunk 1 above)
Comment (Cashmaney) [reply cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs:5]: Ah, cool. The comment was moved by the CLion auto optimize imports thing. With //! it properly leaves it there

File: cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs (hunk 4 above)
Comment (Cashmaney) [reply cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs:282]: > Just a few small notes, I know that this is still not finished :)
> Also, there are a bunch of small formatting adjustments. Were those applied automatically by rustfmt?

yep

Comment (Cashmaney) [reply cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs:282]: Idk it looked fine when it was printed out

File: cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs (hunk 5 above)
Comment (Cashmaney) [reply cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs:232]: I'm just removing the return code it's unused anyway

File: cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs (hunk 4 above)
Comment (reuvenpo) [reply cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs]: ah, cool

File: cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs [hunk 7]
```diff
@@ -1,21 +1,30 @@
-///
+//!
 /// These functions run off chain, and so are not limited by deterministic limitations. Feel free
 /// to go crazy with random generation entropy, time requirements, or whatever else
 ///
```
Comment (reuvenpo) [inline cosmwasm/packages/wasmi-runtime/src/registration/offchain.rs:4]: My intention was that all these lines should be `//!`

**Pull request merged**